connect/connect.go

```diff
@@ -0,0 +1,387 @@
+package connect
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/kyma-incubator/hydroform/connect/types"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+func (c *KymaConnector) Connect(configurationUrl string) error {
+
+	err := c.GetCsrInfo(configurationUrl)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	err = c.GetCertSigningRequest()
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	err = c.GetClientCert()
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	if c.SecureClient == nil {
+		err = c.populateClient()
+		if err != nil {
+			return fmt.Errorf(err.Error())
+		}
+	}
+
+	err = c.PopulateInfo()
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	err = c.WriteClientCertificateToFile()
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	return err
+}
+
```

Comment:
Instead of persisting the Service metadata this function should return the ID. The User of the library can take care of the persistence by him self.